### DISCUSSION TITLE
How can I deploy Frontend and Backend separately?

### DISCUSSION THREAD

**Discussion by magiskboy** [casdoor/casdoor#2402] [Q&A] [answered]
Currently, I want to deploy Frontend in Amazon S3 and Backend in a k8s deployment but I have seen that Casdoor server hosts both FE and BE in a process.

I also see that all of the calls to the backend are using [Setting.ServerUrl](https://github.com/casdoor/casdoor/blob/master/web/src/Setting.js#L31), which is an empty string. It means that FE calls to the same origin.

If we can not deploy on separated places via configuration then do we create a PR for this?

**Comment by nomeguy** [comment, marked as answer]
@magiskboy Casdoor supports using CasWAF (https://github.com/casbin/caswaf) to deploy its frontend to CDN like AWS S3, refer to: https://caswaf.org/
